projects/rawkode.academy/platform/events-service/data-model/integrations/zod.ts

```diff
@@ -0,0 +1,81 @@
+import { eventsTable } from "../schema.ts";
+import { createInsertSchema } from "drizzle-zod";
+import { z } from "zod";
+
+export const EventStatus = z.enum(["scheduled", "cancelled", "rescheduled"]);
+
+export const CreateEvent = createInsertSchema(eventsTable, {
+  id: z.string().min(1).max(255),
+  type: z.enum(["event", "course", "workshop", "live-stream"]),
+  title: z.string().min(1).max(255),
+  description: z.string().min(1).max(255),
+  startDate: z.date(),
+  endDate: z.date(),
+  attendeeLimit: z.number().int().positive().nullable(),
+  url: z.string().url().nullable(),
+  status: EventStatus,
+  originalStartDate: z.date().nullable(),
+  originalEndDate: z.date().nullable(),
+  technologyIds: z.string().transform((val) => {
+    try {
+      const parsed = JSON.parse(val);
+      if (!Array.isArray(parsed)) {
+        throw new Error("Must be a JSON array");
+      }
+      if (!parsed.every((id) => typeof id === "string")) {
+        throw new Error("All IDs must be strings");
+      }
+      return val;
+    } catch (_e) {
+      throw new Error("Invalid JSON array of technology IDs");
+    }
+  }),
+}).refine(
```

Comment:
I don't think we need this. The event page will be available on the website, it's derivative of the event itself. I don't think the event should also understand what a live stream is - this will come from another specific live stream service.